docs/changelog.md

```diff
@@ -5,6 +5,345 @@ description: |
+    such that `[$ floor(x) = lr(floor.l x floor.r) $]`
+  - The [`mat.delim`]($math.mat.delim), [`vec.delim`]($math.vec.delim), and
+    [`cases.delim`]($math.cases.delim) parameters now allow any character that
+    is considered a delimiter or "fence" (e.g. |) by Unicode. The
+    `{delim: "||"}` notation is _not_ supported anymore and should be replaced
+    by `{delim: bar.double}` (**Minor breaking change**)
+  - Added [`underparen`]($math.underparen), [`overparen`]($math.overparen),
+    [`undershell`]($math.undershell), and [`overshell`]($math.underparen)
+  - Added `~` shorthand for `tilde.op` (**Minor breaking change**)
+  - Fixed baseline alignment of equation numbers
+  - Fixed positioning of corner brackets (⌜, ⌝, ⌞, ⌟)
+  - Fixed baseline of large roots
+  - Fixed multiple minor layout bugs with attachments
+  - Fixed that alignment points could affect line height in math
+  - Fixed that spaces could show up between text and invisible elements like
+    [`metadata`] in math
+  - Fixed a crash with recursive show rules in math
+
+- Introspection
+  - Implemented a new system by which Typst tracks where elements end up on the
+    pages. This may lead to subtly different behaviour in introspections.
+    (**Breaking change**)
+  - Fixed various bugs with wrong counter behaviour in complex layout
+    situations, through a new, more principled implementation
+  - Counter updates can now be before the first, in between, and after the last
+    page when isolated by weak page breaks. This allows, for instance, updating
+    a counter before the first page header and background.
+  - Fixed incorrect [`here().position()`]($here) when [`place`] was used in a
+    context expression
+  - Fixed resolved positions of elements (in particular, headings) whose show
+    rule emits an invisible element (like a state update) before a page break
+  - Fixed behaviour of stepping a counter at a deeper level that its current
+    state has
+  - Fixed citation formatting not working in table headers and a few other
+    places
+  - Displaying the footnote counter will now respect the footnote numbering
+    style
+
+- Model
+  - Document set rules do not need to be at the very start of the document
+    anymore. The only restriction is that they must not occur inside of layout
+    containers.
+  - The `spacing` property of [lists]($list.spacing),
+    [enumerations]($enum.spacing), and [term lists]($terms.spacing) is now also
+    respected for tight lists
+  - Tight lists now only attach (with tighter spacing) to preceding paragraphs,
+    not arbitrary blocks
+  - The [`quote`] element is now locatable (can be used in queries)
+  - The bibliography heading now uses `depth` instead of `level` so that its
+    level can still be configured via a show-set rule
+  - Added support for more [numbering] formats: Devanagari, Eastern Arabic, and
+    circled numbers
+  - Added [`hanging-indent`]($heading.hanging-indent) parameter to heading
```

Comment:
Should this be marked as *breaking change* or *minor breaking change*?

Reply:
What does it break?

Reply:
> What does it break?

Headings do have a hanging-indent default value now so multi-line headings will look different. But then again, so will most documents with the new hyphenation rules so one can argue its not necessary to mention this as breaking as then, everything that could ever reflow a paragraph is breaking (plus it doesn't actually break your document, it still compiles).

Reply:
Yeah I'd argue this isn't breaking anything, it's just a changing change—because Typst is so dynamic any change could be observable by code (for example casting a module into a dictionary would yield different results), so I don't think changes that merely affect aesthetics should be warned about in this case. At the end of the day, what counts as a breaking change can only be determined by what behaviour a specific document assumes to be stable